index.js

```diff
@@ -38,10 +38,11 @@ const checkHttp = async (url, timeout) => {
 const getAddress = async hostname => net.isIP(hostname) ? hostname : (await dnsLookupP(hostname)).address;
 
 const isTargetReachable = timeout => async target => {
+	const isHTTP = ['http:', 'https:'].some(prot => target.indexOf(prot) === 0);
```

Comment:
This is overly complicated.

```suggestion
	const isHTTP = target.startsWith('https://') || target.startsWith('http://');
```